modules/apps/object/object-api/src/main/java/com/liferay/object/definition/util/ObjectDefinitionUtil.java

```diff
@@ -161,4 +141,59 @@ private static String _getInvokerFileNameRegex(
 			"User", "L_USER"
 		).build();
 
+	static {
+		if (FeatureFlagManagerUtil.isEnabled("LPD-15729")) {
+			_allowedModifiableSystemObjectDefinitionNames =
```

Comment:
I am not sure if FF check can be included in a single `HashMapBuilder`, so I added it in a separate static block. @dsanz should I ping object folks for review, or is this a normal practice?

Reply:
There is one way that leaves a single `static final` attribute, but pattern is a bit boring as you have to provide the unsafe key supplier once for each map key:
```
		).put(
			() -> {
				if (FeatureFlagManagerUtil.isEnabled("LPD-15729")) {
					return "DSMDateFilter";
				}

				return "FDSDateFilter";
			},
			"/data-set-manager/date-filters"
		).put(
```
This'll work as long as `FeatureFlagManagerUtil` is properly instantiated and working prior to initialize this class, which should be the case as we are invoking it.  

I've seen some `FeatureFlagManagerUtil` usages in static methods but not sure about static initialization of class members, so this does not look common so far. I don't see any impediment however, because `static` blocks come after `static` variables so your map will contain the values at the moment `static` block is reached (see https://www.baeldung.com/java-static-variables-initialization)

Another possibility is to defer addition of the DSM keys to the moment when map is consulted for the first time, i.e, do it in `isAllowedModifiableSystemObjectDefinitionName()` method if keys are not yet there.

Reply:
I'll do a bunch of lambdas in the end, like you suggested above. It's more code, but we don't mess with API.